internal/controller/humiopdfrenderservice_controller.go

```
@@ -446,35 +595,280 @@ func (r *HumioPdfRenderServiceReconciler) getTLSAwareEnvironmentVariables(hprs *
 				Name:  "PDF_RENDER_TLS_KEY_PATH",
 				Value: "/etc/ssl/certs/pdf-render-service/tls.key",
 			},
+		}...)
+	}
+
+	// Log all environment variables for debugging
+	for _, env := range envVars {
+		r.Log.V(1).Info("Environment variable", "name", env.Name, "value", env.Value)
+	}
+
+	return envVars
+}
+
+// validateTLSConfiguration validates the TLS configuration if TLS is enabled
+func (r *HumioPdfRenderServiceReconciler) validateTLSConfiguration(ctx context.Context, hprs *corev1alpha1.HumioPdfRenderService) error {
+	if hprs.Spec.TLS == nil || hprs.Spec.TLS.Enabled == nil || !*hprs.Spec.TLS.Enabled {
+		return nil // TLS not enabled, nothing to validate
+	}
+
+	// Check if certificate secret exists
+	certSecretName := fmt.Sprintf("%s-certificate", hprs.Name)
+	secret := &corev1.Secret{}
+	err := r.Client.Get(ctx, types.NamespacedName{
+		Name:      certSecretName,
+		Namespace: hprs.Namespace,
+	}, secret)
+
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Log warning but don't fail - deployment will fail later if cert is required
+			r.Log.Info("TLS is enabled but certificate secret not found",
+				"secretName", certSecretName,
+				"namespace", hprs.Namespace)
+			return nil
 		}
+		return err
+	}
 
-		// Append TLS env vars to existing env vars
-		if envVars == nil {
-			envVars = tlsEnvVars
-		} else {
-			envVars = append(envVars, tlsEnvVars...)
+	// Validate secret has required keys (tls.crt and tls.key)
```

Should we return an error on these?